src/DataSets/log.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { getInstance as getD2 } from 'd2/lib/d2';
+
+
+async function log(actionName, status, dataset) {
+    // Log the name of the action that has been executed, its status
+    // ("success", "failed"), by whom and on which datasets.
+    const maxLogs = 1e4;
+
+    const d2 = await getD2();
+    const store = await d2.dataStore.get('dataset-configuration');
+
+    const logIndex = await store.get('logIndex').catch(() => 0);
+    const logs = await store.get('logs').catch(() => []);
+
+    const datasets = Array.isArray(dataset) ? dataset : [dataset];
+
+    const newLog = {date: Date(),
```

Comment:
We favour this style for object literals: https://google.github.io/styleguide/jsguide.html#formatting-object-literals